Dialect Sniffing and Bytes support

akariv opened the pull request.

pwalsh commented on goodtables/datatable/datatable.py:
```diff
@@ -118,6 +118,13 @@ def to_textstream(self, data_source):
 
+        elif isinstance(data_source, compat.bytes):
+            self.encoding = self._detect_stream_encoding(data_source)
+            print(self.encoding)
```
can we remove the print function.

pwalsh commented on goodtables/compat.py:
```diff
@@ -43,13 +43,21 @@ def iterenc_utf8(data):
     from urllib import parse
     from urllib.request import urlopen
     from urllib.error import HTTPError
-    csv_reader = csv.reader
```
As this change is only for Python 3 code, we are breaking Py2/3 compatibility. No?

akariv replied:
Yes, I'll add that same functionality for Py2

akariv commented:
Okay @pwalsh , fixed the tests and updated the PR.
Note that some changes were made to adapt to the fact that the `required` constraint was made to be false by default (https://github.com/frictionlessdata/jsontableschema-py/commit/45ebff840eea74a5ca20f80a02a159bacad5330d)

pwalsh commented:
@akariv still a bunch of failing tests.

akariv commented:
Probably because it's using a cached version of `jsontableschema-py` in its
dependencies.
Can you try and invalidate the cache?

On Sun, Mar 27, 2016 at 8:30 AM Paul Walsh <notifications@github.com> wrote:

> @akariv <https://github.com/akariv> still a bunch of failing tests.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/frictionlessdata/goodtables/pull/74#issuecomment-201993134>
>

pwalsh commented:
ok, ready to merge

akariv commented:
@pwalsh I don't have the creds, can you merge?
Also we need to publish to pypi for goodtables.io to build correctly
